File: src/mongoose-dao/data/sync_metadata_datasource.ts
import { SyncMetadata } from "../../core/implementation/metadata/syncable_metadata";
import { SyncMetadataSchema } from "../metadata/sync_metadata_model";
import { SyncalbeObjectDataSource } from "./syncable_object_data_source";

export class SyncMetadataDataSource  extends SyncalbeObjectDataSource<SyncMetadata> {
    constructor() {
        super(SyncMetadataSchema, SyncMetadata.name);
    }
}

File: fast_sync_dotnet_service/fast-sync-entity-framework-dao/data/SyncMetadataDataSource.cs
using fast_sync_core.implementation.metadata;

namespace fast_sync_entity_framework_dao.data
{
    public class SyncMetadataDataSource : SyncableObjectDataSource<SyncMetadata>
    {
        public SyncMetadataDataSource(Func<FastSyncDataContext> classFactory) : base(classFactory) { }
    }
}
